vyper/semantics/types/function.py

```diff
@@ -44,6 +48,7 @@ class KeywordArg(_FunctionArg):
     ast_source: Optional[vy_ast.VyperNode] = None
 
 
+# TODO: refactor this into FunctionT (from an ast) and ABIFunctionT (from json)
```

Comment:
Sounds like https://github.com/vyperlang/titanoboa/pull/108 :smile:

Reply:
yep it's similar :)